src/framework/loader.ts

```diff
@@ -1,7 +1,11 @@
+// Here and in every files we need copyright statements. BSD3 clause license like the WebGPU stuff in the group
+
 import { GroupRecorder } from './logger.js';
 import { IParamsAny, paramsEquals, paramsSupersets } from './params/index.js';
 import { allowedTestNameCharacters, ITestGroup, RunCase, ICaseID } from './test_group.js';
 
+// These data types seem super core so please add more comments
+// Should most members in there be readonly?
```

Comment:
Yes. I forgot for a while that you could make them readonly.

Reply:
Done